src/components/ListV2/ListContext.tsx

```diff
@@ -0,0 +1,107 @@
+import {
+  Dispatch,
+  MutableRefObject,
+  ReactNode,
+  SetStateAction,
+  createContext,
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react'
+
+type ListContextValue<T = Record<string, unknown>> = {
+  template: string
+  selectable?: boolean
+  selectedIds: string[]
+  setSelectedIds: Dispatch<SetStateAction<string[]>>
+  expandedIds: string[]
+  setExpandedIds: Dispatch<SetStateAction<string[]>>
+  data: T[]
+  idKey: string
+  autoClose: boolean
+  disabledRowsRef: MutableRefObject<string[]>
+}
+
+const ListContext = createContext<ListContextValue>({} as ListContextValue)
```

Comment:
```suggestion
const ListContext = createContext<ListContextValue | undefined>(undefined)
```
And then `useListContext` should throw an error if the context is undefined (meaning it has been used without being wrapped in a `ListProvider`)